### ISSUE TITLE
EBU-TT-D encoder with filesystem output carriage mech discards all docs

### ISSUE THREAD

**Issue by nigelmegitt** [ebu/ebu-tt-live-toolkit#454]
Using this config file:

```json
{
  "nodes": {
    "node1": {
      "id": "producer1",
      "type": "simple-producer",
      "show_time": true,
      "sequence_identifier": "TestSequence1",
      "output": {
        "carriage": {
          "type": "direct",
          "id": "pipe1"
        }
      }
    },
    "node2": {
      "id": "resequencer1",
      "type": "resequencer",
      "sequence_identifier": "ReSequenced1",
      "segment_length": "5.0",
      "input": {
        "carriage": {
          "type": "direct",
          "id": "pipe1"
        }
      },
      "output": {
        "carriage": {
          "type": "direct",
          "id": "pipe2"
        }
      }
    },
    "node3": {
      "id": "encoder1",
      "type": "ebuttd-encoder",
      "input": {
        "carriage": {
          "type": "direct",
          "id": "pipe2"
        }
      },
      "output": {
        "carriage": {
          "type": "filesystem"
        }
      }
    }
  }
}
```

produces warnings such as:

```
[WARNING] (2017-07-07 17:10:02,050) in ebu_tt_live.carriage.filesystem[131] - Could not resolve availability time for item in sequence: default. Item discarded.
```

every time a document is due to be emitted, and therefore it emits no documents.

On digging, I see that this is because `EBUTTDEncoder.process_document()` calls `emit_data()` without specifying a `time_base` parameter, and `FilesystemProducerImpl.emit_data()` needs it to get the relevant clock.

**Comment by nigelmegitt** [comment]
Interesting that other nodes than EBU-TT-D Encoder don't suffer from the same thing since none of them specify `time_base`, but I tested with a Retiming Delay node and it seemed fine. Not really sure if there's a deeper problem there.

**Comment by nigelmegitt** [comment]
With #457 this needs to be retested - it should be emitting EBU-TT-D documents formatted with the message file format, regardless of the ability to get an availability time.

**Comment by nigelmegitt** [comment]
I've retested with #457 and this works with commit 2f91f20. Marking as pending PR merge, this can be closed after #457 has been merged.
